src/tdx/azure/azure_tdx_adapter.c

```diff
@@ -385,15 +361,33 @@ int get_td_report(uint8_t *report_data, uint8_t **tpm_report)
 
 	// Get a random number to be appended to the end of file name to make it random
 	int rand_num = rand();
-	char filename[50];
-	// Create file name
-	sprintf(filename, "/tmp/report_azure_%d.txt", rand_num);
+	char filename[50] = {0};
```

Comment:
Suggest using the following code to determine the required buffer size.
size_t nbytes = snprintf(NULL, 0, "/tmp/report_azure_%d.txt", rand_num)

Reply:
fixed in https://github.com/intel/trustauthority-client-for-c/pull/2/commits/a05bc9ae858db41bf6e657a39c480fb8b223febb